src/core/utils/config.tsx

```diff
@@ -2,11 +2,26 @@ import { RootState, useSelector } from "../store";
 import { addConfigEntries } from "../config.slice";
 import withSuffix from "./withSuffix";
 
-export type UseConfigFunction = (key: string) => unknown;
+export type UseConfigFunction = <T>(
+  key: string,
+  options?: { jsonParse?: boolean; stringToArray?: boolean }
+) => string | T[] | undefined;
```

Comment:
Consider if we can provide a more descriptive type for the return value.

It seems as if the method will return a string if options are not defined. It will always return an array of strings if `stringToArray` is set. It will return an array of `T` if `jsonParse` is defined. Also `jsonParse` and `stringToArray` cannot be set at the same time.